adserver/views.py

```diff
@@ -573,6 +581,33 @@ def get_context_data(self, **kwargs):
             "limit": self.LIMIT,
         }
 
+    def get_queryset(self, **kwargs):
```

Comment:
I think the whole point of being able to pass a queryset into the reports is to avoid any filtering logic there. The filtering logic should be on the view. I can see what integrating django-filter would be like.

Reply:
Sure, this is in the view though. I just want another abstraction like `get_filter_queryset` that does similar filtering on date etc in a standard way.

Reply:
To be explicit, I mean this logic:

https://github.com/readthedocs/ethical-ad-server/blob/c7ce6790cdf610c3ba8e3c35dd81c9fe151db3c8/adserver/views.py#L1046-L1062

Reply:
> I just want another abstraction like get_filter_queryset that does similar filtering on date etc

That's exactly what this method does. I don't think I understand what you mean.

Reply:
After discussing we do need a better way to generate these filters. We can use the `get_queryset` method but there still might be an even nicer way.